ddtrace/contrib/psycopg/patch.py

```diff
@@ -10,23 +11,32 @@
 # Original connect method
 _connect = psycopg2.connect
 
-def patch():
+
+def patch(tracer=None):
     """ Patch monkey patches psycopg's connection function
         so that the connection's functions are traced.
     """
-    wrapt.wrap_function_wrapper(psycopg2, 'connect', patched_connect)
-    _patch_extensions() # do this early just in case
+    if getattr(psycopg2, '_datadog_patch', False):
+        return
+    setattr(psycopg2, '_datadog_patch', True)
+
+    wrapt.wrap_function_wrapper(psycopg2, 'connect', functools.partial(patched_connect, tracer=tracer))
+    _patch_extensions(_psycopg2_extensions)  # do this early just in case
+
 
 def unpatch():
-    psycopg2.connect = _connect
+    if getattr(psycopg2, '_datadog_patch', False):
+        setattr(psycopg2, '_datadog_patch', False)
+        psycopg2.connect = _connect
+
 
-def patch_conn(conn):
+def patch_conn(conn, tracer=None, traced_conn_cls=dbapi.TracedConnection):
```

Comment:
gotchya, I see how the global tracer has a `configure` method which will help us do the right thing.  removed